build.gradle

```diff
@@ -75,6 +75,7 @@ dependencies {
 
     provided "junit:junit:4.12", "org.hamcrest:hamcrest-core:1.3"
     compile "org.objenesis:objenesis:2.4"
+    compile 'net.jodah:typetools:0.4.8'
```

Comment:
Looking at the [code](https://github.com/jhalterman/typetools/blob/master/src/main/java/net/jodah/typetools/TypeResolver.java) there seem to be some corner cases in resolving return types.

In this code it is needed to determine a return type of lambda implementing `Consumer` interface. @raphw Maybe you have some code in ByteBuddy which could be used to do that?